### PR TITLE
Fix water tower 2 and dumpster rotation 

### LINKED ISSUE #27: Water Tower 2 is rotated 90 degrees CCW from the prefab's rotation.
The prefab snaps to points and all that, but then is rotated when viewed ingame. Cannot use the turning wheel to rotate it on top of a locomotive either. All other prefabs do not have this rotation.
image

### PR DESCRIPTION
**Pull request by t0stiman**
Fixes #27

### CODE REVIEW

**Comment by t0stiman** [comment]
I broke something, setting this to draft for now

**Comment by t0stiman** [comment]
I forgot to update the MapifyEditor.dll in my Unity project.
afbeelding

File: MapifyEditor/Vanilla/VanillaObject.cs [hunk 1]
```diff
@@ -5,5 +5,7 @@ namespace Mapify.Editor
     public class VanillaObject : MonoBehaviour
     {
         public VanillaAsset asset;
+        public bool keepChildren = true;
+        public Vector3 rotation = Vector3.zero;
```
Comment (Insprill) [inline MapifyEditor/Vanilla/VanillaObject.cs:9]: This should be renamed "rotationOffset" or something similar since it's not setting the exact rotation.

**Review by Insprill — CHANGES_REQUESTED** [review]
Instead of rotating the objects once they get spawned in, I believe the problem is that my models don't match the rotation of the DV models. This would be fairly easy to check by using AssetRipper to export the DV model, then putting both into Blender as-is and seeing if they line up.

**Comment by Insprill** [comment]
Interesting. The model must be rotated under the root in the base game then.

File: MapifyEditor/Vanilla/VanillaObject.cs [hunk 2]
```diff
@@ -1,9 +1,12 @@
 using UnityEngine;
+using UnityEngine.Serialization;
 
 namespace Mapify.Editor
 {
     public class VanillaObject : MonoBehaviour
     {
         public VanillaAsset asset;
+        public bool keepChildren = true;
+        [FormerlySerializedAs("rotation")] public Vector3 rotationOffset = Vector3.zero;
```
Comment (Insprill) [inline MapifyEditor/Vanilla/VanillaObject.cs:10]: Please remove the FormerlySerializedAs attribute and ensure all prefabs are saved with the new name and the correct data.

**Review by Insprill — CHANGES_REQUESTED** [review]

**Review by Insprill — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**